gravitee-rest-api-management/gravitee-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/RoleResource.java

```diff
@@ -48,7 +48,7 @@
     @Context
     private ResourceContext resourceContext;
 
-    @Autowired
+    @Inject
     private RoleService roleService;
 
```

Comment:
Maybe you could declare 2 @PathParam here ?
```
    @PathParam("scope")
    @ApiParam(name = "scope", hidden=true)
    private String scope;

    @PathParam("role")
    @ApiParam(name = "role", required=true)
    private String role;
```
WDYT ?

Reply:
Since `RoleResource` is "parent" of `RoleUsersResource`, but `RoleUsersResource` has multiple parents, we cannot migrate the method params to properties.

Hierarchie:

1. `OrganizationResource` -> `EnvironmentsResource` -> `EnvironmentResource` -> `ConfigurationResource` -> `RoleScopesResource` -> `RoleScopeResource` -> _`RoleResource`_ -> **`RoleUsersResource`**
2. `OrganizationResource` -> **`RoleUsersResource`**

Reply:
Ok with this. This problem will be solved when deprecated paths will removed. This is plan is a future 4.x version.
See : https://github.com/gravitee-io/release/tree/master/upgrades/3.x/3.3.0#management-api